skbio/tree/_tree.py

```diff
@@ -2327,7 +2327,11 @@ def _set_max_distance(self):
                 if len(n.children) == 1:
                     raise TreeError("No support for single descedent nodes")
                 else:
-                    tip_info = [(max(c.MaxDistTips), c) for c in n.children]
+                    tip_info = []
+                    for c in n.children:
+                        (left_d, _), (right_d, _) = c.MaxDistTips
```

Comment:
I got part way through fixing this before I saw this PR. This lets you do it without creating a new list. Doesn't matter to me which way we solve this, just wanted to share this since I started on it. 

```
                    tip_info = []
                    for c in n.children:
                        tip_info.append((max(c.MaxDistTips, key=lambda x: x[0]), c))
```